### DISCUSSION TITLE
Filename OrderOptionType

### DISCUSSION THREAD

**Discussion by jj-gh** [fluttercandies/flutter_photo_manager#972] [Q&A]
### Platforms

Android, iOS

### Description

It would be nice if we could sort by filename instead of just createDate and updateDate.

**Comment by CaiJingLong** [comment]
In Android, it can be implemented using the `CustomFilter`.
But in iOS/macOS there is no way, because iOS does not natively provide a title attribute for `PHAsset`.

---

For android: 
- [CustomFilter](https://pub.dev/documentation/photo_manager/latest/photo_manager/CustomFilter-class.html)
- [title](https://pub.dev/documentation/photo_manager/latest/photo_manager/AndroidMediaColumns/title.html)
- [OrderByItem](https://pub.dev/documentation/photo_manager/latest/photo_manager/OrderByItem-class.html)

**Reply by jj-gh** [reply]
Sorry, I can't seem to make it work. I'm trying to do it on Android.

This is how I'm sorting the albums:
```dart
final albums = await PhotoManager.getAssetPathList(
  hasAll: false,
  filterOption: CustomFilter.sql(
    where: '',
    orderBy: [
      OrderByItem(CustomColumns.android.title, true),
    ],
  ),
);
```
And how I get an album's media list:
```dart
final mediaList = await albums[0].getAssetListPaged(page: page, size: size);
```
I assumed the sorting should also apply to AssetPathEntity.getAssetListPaged, right? Or does it only apply to PhotoManager.getAssetPathList?

**Comment by jj-gh** [comment]
Turns out it was actually working. The issue was on another part of my code and got confused about the output during testing. Thanks for the assist and sorry for the inconvenience.
